modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2249,144 +2324,179 @@ private Expression<?>[] _getSelectExpressions(
-							objectDefinition.getPKObjectFieldDBColumnName(),
-							relatedObjectDefinition.
-								getPKObjectFieldDBColumnName(),
-							objectRelationship.getDBTableName());
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn2 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn2();
-
-				joinStep = joinStep.innerJoinON(
-					dynamicObjectRelationshipMappingTable,
-					primaryKeyColumn2.eq(
-						relatedDynamicObjectDefinitionTable.
-							getPrimaryKeyColumn()));
-
-				Column<DynamicObjectRelationshipMappingTable, Long>
-					primaryKeyColumn1 =
-						dynamicObjectRelationshipMappingTable.
-							getPrimaryKeyColumn1();
-
-				predicate = primaryKeyColumn1.eq(
-					dynamicObjectDefinitionTable.getPrimaryKeyColumn());
-			}
-
-			List<String> oDataFilterStrings = TransformUtil.transform(
-				(List<ObjectFilter>)objectFieldSettingsValues.get("filters"),
-				objectFilter -> {
-					if (StringUtil.equals(
-							objectFilter.getFilterType(),
-							ObjectFilterConstants.TYPE_CURRENT_USER)) {
-
-						objectFilter.setJSON(
-							JSONUtil.put(
-								"currentUserId",
-								PrincipalThreadLocal.getUserId()
-							).toString());
-					}
+				List<String> oDataFilterStrings = TransformUtil.transform(
+					(List<ObjectFilter>)objectFieldSettingsValues.get(
+						"filters"),
+					objectFilter -> {
+						if (StringUtil.equals(
+								objectFilter.getFilterType(),
+								ObjectFilterConstants.TYPE_CURRENT_USER)) {
+
+							objectFilter.setJSON(
+								JSONUtil.put(
+									"currentUserId",
+									PrincipalThreadLocal.getUserId()
+								).toString());
+						}
+
+						ObjectFilterParser objectFilterParser =
+							_objectFilterParserServiceRegistry.
+								getObjectFilterParser(
+									objectFilter.getFilterType());
 
-					ObjectFilterParser objectFilterParser =
-						_objectFilterParserServiceRegistry.
-							getObjectFilterParser(objectFilter.getFilterType());
+						return objectFilterParser.parse(objectFilter);
+					});
 
-					return objectFilterParser.parse(objectFilter);
-				});
+				for (String oDataFilterString : oDataFilterStrings) {
+					predicate = predicate.and(
+						_filterPredicateFactory.create(
+							oDataFilterString,
+							relatedObjectDefinition.getObjectDefinitionId()));
+				}
 
-			for (String oDataFilterString : oDataFilterStrings) {
-				predicate = predicate.and(
-					_filterPredicateFactory.create(
-						oDataFilterString,
-						relatedObjectDefinition.getObjectDefinitionId()));
+				selectExpressions.add(
+					DSLQueryFactoryUtil.scalarSubDSLQuery(
+						joinStep.where(predicate),
+						DynamicObjectDefinitionTable.getJavaClass(
+							objectField.getDBType()),
+						objectField.getName(),
+						DynamicObjectDefinitionTable.getSQLType(
+							objectField.getDBType())));
 			}
+			else if (objectField.compareBusinessType(
+						ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
```

Comment:
One time, Brian asked me to order alphabetically the conditions between IF/ELSE IF. So, this condition should be after comparing with Aggregation field.

I don't know if you would like to do this refactor.

Reply:
Hey @marcelabc I didn't understand your point. I verified it here and it seems to be already sorted.

`if (objectField.compareBusinessType(
					ObjectFieldConstants.BUSINESS_TYPE_AGGREGATION)) {
				...
}
else if (objectField.compareBusinessType(
		ObjectFieldConstants.BUSINESS_TYPE_FORMULA)) {
				...
}`

Reply:
Sorry, I misunderstood the code indentation.